### ISSUE TITLE
ETL node awareness abstraction

### ISSUE THREAD

**Issue by godexsoft** [XRPLF/clio#1595]
## Summary
Part of #1594 

Create a layer of abstraction for writer handover process. 
The logic does not yet need to be implemented in any different way from the existing one. This just prepares the grounds for future node awareness protocols. Related to #1183  

## Motivation
In #1183 Clio should gain knowledge about its nodes in the cluster. ETL needs to be prepared to support this. 

## Solution
Create an abstraction and implement using the existing approach.

**Comment by kuznetsss** [comment]
I don't think this should be a part of ETL. Potentially any part of Clio should be able to use node communication API so I would suggest to separate it from ETL.
